Spigot-Server-Patches/0399-Optimize-Hoppers.patch

```diff
@@ -454,7 +457,7 @@ index a0a3adac3e2bc939b1809c5587929f674f4318a5..20df9bd21d0e4d2579d05d79672da2eb
          }
  
 -        if (object == null) {
-+        if (object == null && (!optimizeEntities || !org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(block).isOccluding())) { // Paper
++        if (object == null && (!optimizeEntities || !org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(block).isOccluding() || !world.paperConfig.hoppersIgnoreOccludingBlocks)) { // Paper
```

Comment:
Isn't this backwards? The check should also be flipped: checking a known boolean is always faster than getting the material of a block.

Reply:
Sorry, I think I had my priorities swapped when I wrote this originally. I approached this from the angle of how many people it would affect and I was imagining that only a small number of servers would change it from the default. It is true though that for this servers that do change this it would be much more efficient to have the check in front, with a negligible difference for those that don't. Made the requested changes.